plugins/azure-devops/src/hooks/useGitTags.ts

```diff
@@ -20,18 +20,21 @@ import { Entity } from '@backstage/catalog-model';
 import { azureDevOpsApiRef } from '../api';
 import { useApi } from '@backstage/core-plugin-api';
 import useAsync from 'react-use/lib/useAsync';
-import { useProjectRepoFromEntity } from './useProjectRepoFromEntity';
+import { getAnnotationValuesFromEntity } from '../utils';
 
 export function useGitTags(entity: Entity): {
   items?: GitTag[];
   loading: boolean;
   error?: Error;
 } {
   const api = useApi(azureDevOpsApiRef);
-  const { project, repo } = useProjectRepoFromEntity(entity);
+  const { project, repo } = getAnnotationValuesFromEntity(entity);
 
-  const { value, loading, error } = useAsync(() => {
-    return api.getGitTags(project, repo);
+  const { value, loading, error } = useAsync(async () => {
+    if (repo) {
+      return await api.getGitTags(project, repo);
+    }
+    return undefined;
```

Comment:
Yes, I can do that. I'm having a bit of a hard time with this. Any chance you know of some examples of testing hooks?

Reply:
https://github.com/backstage/backstage/blob/master/plugins/catalog-react/src/hooks/useEntity.test.tsx#L32

Reply:
I've added tests for all the hook I changed to use `getAnnotationValuesFromEntity`. I also took out this if check as if this is missing it would already have thrown an error before this.

One thing I ran into with the test was that I needed to use a `tsx` extension but my hook are just `ts`, feel like I might be missing something obvious. Advice welcome